### PR TITLE
feat: nft watchlist

### PR DESCRIPTION
**Pull request by phucledien**
**What does this PR do?**

-   [x] View/add/remove nft watchlist
-   [x] Support slash command

**How to test**

-  `$wl add-nft neko` | `/watchlist add-nft neko`
-  `$wl remove neko` | `/watchlist remove-nft neko`
- `$wl view` | `/watchlist view`

**Media (Loom or gif)**

CleanShot 2022-09-28 at 19 50 27@2x

CleanShot 2022-09-28 at 19 51 10@2x

CleanShot 2022-09-28 at 19 51 36@2x

CleanShot 2022-09-28 at 19 51 51@2x

### CODE REVIEW

File: src/commands/defi/watchlist_slash/view.ts [hunk 1]
```diff
@@ -16,8 +25,16 @@ import {
 } from "utils/canvas"
 import CacheManager from "utils/CacheManager"
 import { APIError } from "errors"
+import { MessageComponentTypes } from "discord.js/typings/enums"
 
 let fontRegistered = false
+let currentView = "token"
```
Comment (anhnh12) [inline src/commands/defi/watchlist_slash/view.ts:31]: should be local var

**Review by anhnh12 — CHANGES_REQUESTED** [review]

File: src/commands/defi/watchlist/add-nft.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    const opt = (
+      collection: ResponseCollectionSuggestions
+    ): MessageSelectOptionData => ({
+      label: `${collection.name} (${collection.symbol})`,
+      value: `${collection.symbol}_${collection.address}_${collection.chain}_${msg.author.id}`,
+    })
+    const selectRow = composeDiscordSelectionRow({
+      customId: "watchlist_nft_selection",
+      placeholder: "Make a selection",
+      options: suggestions.map(opt),
```
Comment (anhnh12) [inline src/commands/defi/watchlist/add-nft.ts:82]: `suggestions` can be empty which may cause error

**Review by anhnh12 — CHANGES_REQUESTED** [review]

File: src/commands/defi/watchlist_slash/view.ts (hunk 1 above)
Comment (phucledien) [reply src/commands/defi/watchlist_slash/view.ts]: Removed the var

File: src/commands/defi/watchlist/add-nft.ts (hunk 2 above)
Comment (phucledien) [reply src/commands/defi/watchlist/add-nft.ts:96]: I did handle the empty case

**Review by anhnh12 — APPROVED** [review]

**Review by tuanddd — APPROVED** [review]

**Review by trkhoi — APPROVED** [review]

**Pull request merged**